test/test_autograd.py

```diff
@@ -2536,6 +2536,20 @@ def test_profiler(self):
             self.assertEqual(info.name, expected_name)
             last_end = info.cpu_interval.end
 
+    def test_record_function_callbacks(self):
+        x = torch.randn(10, 10)
+        with profile() as p:
+            rf = torch.autograd._RecordFunction()
+            torch.autograd._run_before_callbacks(rf, "foo")
+            y = x * 2 + 4
+            # ensure that we run destructor for RecordFunction, which invokes
+            # end callbacks
+            del rf
+            gc.collect()
```

Comment:
Hmm, so if I understand correctly, I thought that `del` would just unbind the name rf from the object, and it's not necessary that the object would be destroyed immediately when we call del. I wanted to force the destructrion so that the `RecordFunction` destructor runs, hence the gc.collect which should recognize there are no more references and destroy the object.

Reply:
Since we only use `cpython` that does refcounting. All the objects that are not in cycles will be deleted as soon as the last reference to it is deleted. So it won't be needed here.
It is better to remove it from here as you don't actually want `rf` to be in a cycle and want to catch in the future if it is added to a cycle by mistake.